src/status_im/contexts/profile/push_notifications/effects.cljs

```diff
@@ -1,60 +1,130 @@
+  (-> (attempt-enable-notifications)
+      (promesa/then
+       (fn [_permissions]
+         (pn-remote-token/request-fcm-token {})))
+      (promesa/then
+       (fn [{:keys [fcm-token]}]
+         (logger/log-debug "registered fcm token for status news"
+                           {:fcm-token fcm-token})))
+      (promesa/catch
+        (fn [error]
+          (logger/log-debug "failed to register fcm token for status news"
+                            {:error error})))))
 
-(defn disable-ios-notifications
+(defn disable-news-notifications
+  []
+  (pn-remote-token/revoke-fcm-token {}))
+
+(defn enable-chat-notifications-ios
   []
-  (pn-ios/abandon-permissions)
+  (-> (attempt-enable-notifications)
+      (promesa/then
+       (fn [_permissions]
+         (pn-remote-token/request-apns-token {})))
+      (promesa/then
+       (fn [{:keys [apns-token]}]
+         ;; NOTE(seanstrom): We're only registering iOS devices for remote chat push notifications
+         ;; because Android will use local notifications from the background for chat
+         ;; notifications. This is a temporary workaround that attempts to prevent Android users
+         ;; from receiving two push notifications for each chat message when the app is running
+         ;; the background. This workaround can be removed after we support remote push
+         ;; notifications with message previews that decrypt locally.
+         (when platform/ios?
+           (rf/dispatch [:push-notifications/registered-for-push-notifications apns-token]))))))
+
+(defn disable-chat-notifications-ios
+  []
+  (pn-permissions/release-notification-permissions)
   (rf/dispatch [:push-notifications/unregistered-from-push-notifications]))
 
-(defn enable-android-notifications
+(defn enable-ios-notifications
+  [{:keys [enable-chat-notifications? enable-news-notifications?]}]
+  (when enable-chat-notifications?
+    (enable-chat-notifications-ios))
+  (when enable-news-notifications?
+    (enable-news-notifications)))
+
+(defn disable-ios-notifications
+  [{:keys [disable-chat-notifications? disable-news-notifications?]}]
+  (when disable-chat-notifications?
+    (disable-chat-notifications-ios))
+  (when disable-news-notifications?
+    (disable-news-notifications)))
+
+(defn enable-chat-notifications-android
   []
-  (native-module.pn/create-channel
-   {:channel-id   "status-im-notifications"
-    :channel-name "Status push notifications"})
-  (native-module.pn/enable-notifications))
+  (-> (attempt-enable-notifications)
+      (promesa/then
+       (fn [_permissions]
+         (native-module.pn/create-channel
+          {:channel-id   "status-im-notifications"
+           :channel-name "Status push notifications"})
+         (native-module.pn/enable-notifications)))))
+
+(defn enable-android-notifications
+  [{:keys [enable-news-notifications? enable-chat-notifications?]}]
+  (when (and enable-news-notifications? (not config/fdroid?))
+    (enable-news-notifications))
```

Comment:
Are we also implementing information message/popup for fdroid builds? (Like this feature not available for this build etc.)

Reply:
Yes that's correct, this feature for news-notifications would not be available on FDroid builds (at least for now).